### PR TITLE
[MOB-32] Search filters

### PR DESCRIPTION
**Pull request by ameer2468**
Expect some minor tweaks soon in regards to the UI.

https://github.com/spacedriveapp/spacedrive/assets/33054370/c03981df-27d9-4d49-9bd0-15ac2e5053a7

### CODE REVIEW

File: apps/mobile/src/components/filters/FiltersList.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		{ name: 'Extension', icon: Textbox, component: Extension },
+		{
+			name: 'Hidden',
+			icon: SelectionSlash
+		}
+	] satisfies {
+		name: Capitalize<SearchFilters>;
+		icon: FunctionComponent<IconProps>;
+		component?: FunctionComponent<any>;
+	}[];
```
Comment (utkubakir) [inline apps/mobile/src/components/filters/FiltersList.tsx:44]: can be moved outside

File: apps/mobile/src/components/filters/Locations.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+			<View>
+				<Fade color="mobile-screen" width={30} height="100%">
+					<VirtualizedListWrapper horizontal>
+						<FlatList
+							data={locations}
+							renderItem={({ item }) => <LocationFilter data={item} />}
+							contentContainerStyle={tw`pl-6`}
+							numColumns={locations && Math.ceil(Number(locations.length) / 2)}
+							extraData={searchStore.filters.locations}
+							key={locations ? 'locationsSearch' : '_'}
```
Comment (utkubakir) [inline apps/mobile/src/components/filters/Locations.tsx:42]: why do we need key here?

File: apps/mobile/src/components/filters/SaveAdd.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+				style={twStyle(`h-10 flex-1`, {
+					opacity: searchStore.disableActionButtons ? 0.5 : 1
+				})}
+				variant="accent"
+				onPress={() => {
+					searchStore.applyFilters();
+					navigation.navigate('SearchHome');
+				}}
+			>
+				<Text style={tw`font-medium text-ink`}>+ Add filters</Text>
```
Comment (utkubakir) [inline apps/mobile/src/components/filters/SaveAdd.tsx:40]: Instead of +, let's use Icon here

File: apps/mobile/src/components/filters/Tags.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+			<View>
+				<Fade color="mobile-screen" width={30} height="100%">
+					<VirtualizedListWrapper horizontal>
+						<FlatList
+							data={tagsData}
+							renderItem={({ item }) => <TagFilter tag={item} />}
+							contentContainerStyle={tw`pl-6`}
+							extraData={searchStore.filters.tags}
+							numColumns={tagsData && Math.ceil(Number(tagsData.length ?? 0) / 2)}
+							key={tagsData ? 'tagsSearch' : '_'}
```
Comment (utkubakir) [inline apps/mobile/src/components/filters/Tags.tsx:42]: why key

File: apps/mobile/src/navigation/SearchStack.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import SearchScreen from '~/screens/search';
+import FiltersScreen from '~/screens/search/Filters';
+
+const Stack = createStackNavigator<SearchStackParamList>();
+
+export default function SearchStack() {
+	return (
+		<Stack.Navigator initialRouteName="SearchHome">
+			<Stack.Screen
+				name="SearchHome"
```
Comment (utkubakir) [inline apps/mobile/src/navigation/SearchStack.tsx:13]: just `Home` works better

File: apps/mobile/src/components/filters/Tags.tsx (hunk 4 above)
Comment (ameer2468) [reply apps/mobile/src/components/filters/Tags.tsx:42]: numColumns re-renders and the key is required to do just that, React gave a clear error about this one.

File: apps/mobile/src/components/filters/Locations.tsx (hunk 2 above)
Comment (ameer2468) [reply apps/mobile/src/components/filters/Locations.tsx:42]: Same as other

File: apps/mobile/src/components/filters/Tags.tsx (hunk 4 above)
Comment (utkubakir) [reply apps/mobile/src/components/filters/Tags.tsx:42]: makes sense but why is it conditional? how about just `tagsSerch`?

Comment (ameer2468) [reply apps/mobile/src/components/filters/Tags.tsx:42]: Screenshot 2024-03-10 at 9 06 56 PM

Comment (utkubakir) [reply apps/mobile/src/components/filters/Tags.tsx:42]: good to know! thanks for bearing with me @ameer2468 :D

**Review by brxken128 — APPROVED** [review]
Looks good to me! I couldn't see anything amiss :)

**Pull request merged**